docs/iot/tutorials/gpio-input.md

```diff
@@ -0,0 +1,138 @@
+
+In this tutorial, you'll use .NET and your Raspberry Pi's GPIO pins to detect the opening and closing of a circuit.
+
+## Prerequisites
+
+- [!INCLUDE [prereq-rpi](../includes/prereq-rpi.md)]
+- Jumper wires
+- Breadboard (optional)
+- Raspberry Pi GPIO breakout board (optional)
+- [!INCLUDE [tutorial-prereq-dotnet](../includes/tutorial-prereq-dotnet.md)]
+
+[!INCLUDE [ensure-ssh](../includes/ensure-ssh.md)]
+
+## Prepare the hardware
+
+Use the hardware components to build the circuit as depicted in the following diagram:
+
+:::image type="content" source="https://via.placeholder.com/400x200.png" alt-text="A Fritzing diagram showing a circuit that connects a ground pin to pin 21." lightbox="https://via.placeholder.com/800x400.png":::
+
+The image above depicts a direct connection between a ground pin and pin 21.
+
+> [!TIP]
+> The diagram uses a breadboard and GPIO breakout for illustrative purposes, but feel free to just jumper a ground pin and pin 21 on the Raspberry Pi.
+
+[!INCLUDE [tutorial-rpi-gpio](../includes/tutorial-rpi-gpio.md)]
+
+## Create the app
+
+Complete the following steps in your preferred development environment:
+
+1. Create a new .NET Console App using either the [.NET CLI](../../core/tools/dotnet-new.md) or [Visual Studio](../../core/tutorials/with-visual-studio.md). Name it *InputTutorial*.
+
+    ```dotnetcli
+    dotnet new console -o InputTutorial
+    ```
+
+1. [!INCLUDE [tutorial-add-packages](../includes/tutorial-add-packages.md)]
+1. Replace the contents of *Program.cs* with the following code:
+
+    :::code language="csharp" source="~/iot-samples/tutorials/InputTutorial/Program.cs" :::
+
+    In the preceding code:
+
+    - A [using declaration](../../csharp/whats-new/csharp-8.md#using-declarations) creates an instance of `GpioController`. The `using` declaration ensures the object is disposed and hardware resources are released properly.
+    - GPIO pin 21 is opened with  `PinMode.InputPullUp`.
+        - This opens the pin with a *PullUp* resistor engaged. In this mode, when the pin is connected to ground, it will return `PinValue.Low`. When the pin is disconnected from ground and the circuit is open, the pin returns `PinValue.High`.
+    - The initial status is written to a console using a ternary expression. The pin's current state is read with `Read()`. If it's `PinValue.High`, it writes the `alert` string to the console. Otherwise, it writes the `ready` string.
+    - `RegisterCallbackForPinValueChangedEvent()` registers an anonymous callback function for both the `PinEventTypes.Rising` and `PinEventTypes.Falling` events on the pin. These events correspond to pin states of `PinValue.High` and `PinValue.Low`, respectively.
+    - The callback function delegates to a method called `onPinEvent`. `onPinEvent` uses another ternary expression that also writes the corresponding `alert` or `ready` strings.
```

Comment:
```suggestion
    - The callback function delegates to a method called `OnPinEvent`. `OnPinEvent` uses another ternary expression that also writes the corresponding `alert` or `ready` strings.
```

Consider renaming this as all other C# methods.

Reply:
Wait... C# method names should be Pascal Case? 

Why have I been using Camel Case? My entire life is a lie.

Reply:
No, it turns out, I knew that. Then why does it suddenly seem wrong?

I'm scared. Hold me.